### PR TITLE
Fix handle inheritance for Python2 (more appropriate and safe version of #1138 hack, fixes #1136)

### LINKED ISSUE #1136: [Windows] POWERSHELL_MANAGER is never closed explicitly
scapy-2.4.0rc4.dev58-py2.7
Windows 7

POWERSHELL_MANAGER isn't closed in arch/windows/__init__.py

That's a problem, because subprocess used inherits open files. The files can't be deleted without waiting for scapy to end (sleep-wait-repeat).

I've added POWERSHELL_MANAGER.close() near the bottom of __init__.py and it solved the problem.

Steps to reproduce:
1. Create a file.
2. Replace sys.stdout/stderr with it (I'm doing a tee)
3. Import scapy
4. Try to remove created file after closing it

Observed behaviour:
- exception - file in use, can't be deleted

Expected behaviour:
- file is deleted just fine

Reproducible:
- always

### PR DESCRIPTION
**Pull request by karpierz**
Scapy Version: 2.4.0rc5-30
System: Windows7/Windows10
Python Version: 2.7.14/3.4.4/3.6.4

### CODE REVIEW

**Comment by karpierz** [comment]
Manual tests (atomatic tests cannot be performed because it is launched _after_ scapy starts):

```
import os
import scapy.arch.windows

def test():
    open("ZZZ", "w").write("ddddd")
    with open("ZZZ") as f:
        POWERSHELL_PROCESS = scapy.arch.windows._PowershellManager()
    os.remove("ZZZ")

print("\nWith prevent file handles inheritance")
test()
print("OK")

print("\nWithout prevent file handles inheritance")
scapy.arch.windows._suppress_handles_inheritance = lambda r=1000: []
test()
print("OK")
```

-------------------------------------

**Comment by karpierz** [comment]
PR is ready for merge.

**Comment by gpotter2** [comment]
Sorry I don't understand what's not working.
Could you please provide a sample of something that crashes with current code ?

I tested the scripts you provided and could not make it crash :/

**Comment by karpierz** [comment]
:> I tested the scripts you provided and could not make it crash :/
Because I'v changed a name of the function but you - I supose - use a master - not my branch.
Just now I'v restored original: in the repo and in the example test. Please try again. 

I cannot attach the code because it is production code of out customers - code that opens a lot of files and redirecting stdout/stderr by the different way.
Simply: your patch changes inheritance mode for handlers  and does not restored it (my only realized such weakness).
Missing of restoring inherotance mode may and have influence for other "foreign" subprocesses.
It is even not compatible with Py3 where handle inheritance protection is being implemented by other way than change inheritance mode - by the appropraiate parameter of CreateProcess WINAPI function.

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -83,55 +83,89 @@ def _windows_title(title=None):
...
+    from ctypes import windll, wintypes
+
+    HANDLE_FLAG_INHERIT = 0x00000001
+
+    for osf_handle in handles:
+        try:
+            handle = wintypes.HANDLE(osf_handle)
             mask = wintypes.DWORD(HANDLE_FLAG_INHERIT)
-            flags = wintypes.DWORD(0)
+            flags = wintypes.DWORD(HANDLE_FLAG_INHERIT)
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: Please leave 1

File: scapy/arch/windows/__init__.py [hunk 2]
```diff
@@ -83,55 +83,89 @@ def _windows_title(title=None):
...
     """HACK: python 2.7 file descriptors.
 
     This magic hack fixes https://bugs.python.org/issue19575
     by suppressing the HANDLE_FLAG_INHERIT flag to a range of
     already opened file descriptors.
     """
-    # See https://github.com/secdev/scapy/issues/1136
+    # Fix https://bugs.python.org/issue19575
+    # and https://github.com/secdev/scapy/issues/1136
+    # Bug was fixed on python 3.4+
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: Could you remove the duplicated line (see above), and merge this with the docstring ?

File: scapy/arch/windows/__init__.py (hunk 1 above)
Comment (karpierz) [reply scapy/arch/windows/__init__.py]: Better is HANDLE_FLAG_INHERITED.
Take a look at:
https://msdn.microsoft.com/pl-pl/library/windows/desktop/ms724935(v=vs.85).aspx

Comment (gpotter2) [reply scapy/arch/windows/__init__.py]: ok

**Comment by karpierz** [comment]
Still in progress! Do not merge.

**Comment by karpierz** [comment]
Closed permanently  due to accident (forced push).
See a renew as #1236.

**Pull request closed without merging**